### PR TITLE
[Doc] Integrate knowledge base in docs

### PR DESCRIPTION
**Pull request by hatala91**
## Description

With this pull request, I propose to include the existing knowledge base articles in our [documentation](https://pytorch.org/rl/).

Since Sphinx doesn't natively support files outside the source directory, I am generating one file per knowledge base entry to reference it using the `rst`/ `doctool` include directive and `myst` for parsing markdown.

https://user-images.githubusercontent.com/19345882/198412236-3991f8ce-5a31-4499-95c9-75999d97ff3f.mov

## Motivation and Context

Making the knowledge base more accessible by including it in the docs.

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/source/knowledge_base.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from pathlib import Path
+from typing import List
+
+FILE_DIR = os.path.dirname(__file__)
+GEN_DIR = "reference/generated/knowledge_base"
+
+
+def _get_file_content(name: str) -> List[str]:
+    """
+    To include markdown files from the knowledge graph in Sphinx, these are linked using the rst include command. This
```
Comment (vmoens) [inline docs/source/knowledge_base.py:11]: The proper way of writing docstrings should be without a new line after the triple quote, and a short description first
```python
def fun():
    """This function does X.

    A way more elaborate explanation.
    """
    return foo
```

File: docs/source/knowledge_base.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        "\n",
+        f".. include:: ../../../../../knowledge_base/{name}.md\n",
+        "   :parser: myst_parser.sphinx_\n",
+        "\n",
+    ]
+
+
+def generate_knowledge_base_references(knowledge_base_path: str) -> None:
+    """
+    Sphinx natively doesn't support adding files from outside it's root directory. To include the knowledge base in
```
Comment (vmoens) [inline docs/source/knowledge_base.py:29]: same as above.

it's -> its

**Review by vmoens — APPROVED** [review]
Wonderful! Just a couple of nits

File: docs/source/knowledge_base.py (hunk 1 above)
Comment (hatala91) [reply docs/source/knowledge_base.py]: You are absolutely right, thanks!

File: docs/source/knowledge_base.py (hunk 2 above)
Comment (hatala91) [reply docs/source/knowledge_base.py]: Good catch 👀

**Pull request merged**